benchmarks/bvh_driver/benchmark_registration.hpp

```diff
@@ -196,6 +196,19 @@ struct CountCallback
   }
 };
 
+template <typename DeviceType>
+struct CountCallbackKernelQuery
+{
+  Kokkos::View<int *, DeviceType> count_;
+
+  template <typename Query, typename Primitive>
+  KOKKOS_FUNCTION void operator()(Query const &query, Primitive const &) const
+  {
+    auto const i = ArborX::getData(query);
+    Kokkos::atomic_increment(&count_(i));
+  }
+};
+
```

Comment:
Why not modify the existing `CountCallback`? Just switch the second argument from hardcoded `int` to `Value` template argument.